src/main/java/com/joy/joyfillandroid/Tablefield/MobileSDKField.kt

```diff
@@ -0,0 +1,401 @@
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+        componentLayout.orientation = LinearLayout.VERTICAL
+        componentLayout.layoutParams = componentLayoutParams
+
+        val view = LinearLayout(context)
+        val viewLayoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT,
+        )
+
+        view.layoutParams = viewLayoutParams
+        view.orientation = VERTICAL
+        view.background = layoutBackground()
+
+        val scrollView = NestedScrollView(context)
+        val scrollViewParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT
+        )
+        scrollViewParams.setMargins(0, dpToPx(context, 10), 0, dpToPx(context, 10))
+        scrollView.layoutParams= scrollViewParams
+        scrollView.setVerticalScrollBarEnabled(false)
+        scrollView.overScrollMode= View.OVER_SCROLL_NEVER
+        scrollView.fullScroll(ScrollView.FOCUS_DOWN)
+
+        scrollView.addView(componentLayout)
+        view.addView(scrollView)
+        addView(view)
+        makeApiCall()
+    }
+
+    //Mark: Function for API Call
+    fun makeApiCall() {
+        val url = "https://api-joy.joyfill.io/v1/documents/"+ SharedData.joyfillDocID
+        val request = Request.Builder()
+            .url(url)
+            .addHeader("Authorization", SharedData.authorizedToken)
+            .addHeader("Content-Type", "application/json")
+            .build()
+
+        client.newCall(request).enqueue(object : okhttp3.Callback {
+            @RequiresApi(Build.VERSION_CODES.P)
+            override fun onResponse(call: okhttp3.Call, response: okhttp3.Response) {
+                val responseBody = response.body?.string()
+                val outputJsonString =  JSONObject(responseBody)
+                val fileArray = outputJsonString.getJSONArray("files")
+                val fields = outputJsonString.getJSONArray("fields")
+
+                for (i in 0 until fileArray.length()){
+                    val pages = fileArray.getJSONObject(i).getJSONArray("pages")
+                    val views = fileArray.getJSONObject(i).getJSONArray("views")
+
+                    if (views.length()>0){
+                        // For Mobile View
+                        for (j in 0 until views.length()){
+                            val view_pages = views.getJSONObject(j).getJSONArray("pages")
+                            for (k in 0 until view_pages.length()){
+                                val fieldPositions =view_pages.getJSONObject(k).getJSONArray("fieldPositions")
+                                for (l in 0 until fieldPositions.length()){
+                                    val component_Type = fieldPositions.getJSONObject(l).get("type")
+                                    val Y_Position =  fieldPositions.getJSONObject(l).getDouble("y")
```

Comment:
Is there a reason for such varying variable `val` naming? You can see just in this one area you have 3 different naming conventions `view_pages`, `fieldPositions`, `Y_Position` ect... please look through all the code and make sure naming of variables and functions stays consistent.

While speed is great for us we still value heavily quality as we plan to bring in many more devs so keeping code quality clear and commented well is important.

Reply:
This is Kotlin’s standard way to create variable with val and I followed the camel casing approach for naming conventions.